avocado/core/app.py

```diff
@@ -38,6 +39,16 @@ def __init__(self):
         # Catch all libc runtime errors to STDERR
         os.environ['LIBC_FATAL_STDERR_'] = '1'
 
+        def sigterm_handler(signum, frame):     # pylint: disable=W0613
+            children = process.get_children_pids(os.getpid(), recursive=True)
+            for child in children:
+                try:
+                    os.kill(int(child), signal.SIGTERM)
```

Comment:
One more thing, I'd recommend using `SIGKILL` as this is basically an emergency and we want to destroy as much processes as possible.